airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_application_fees.py

```diff
@@ -0,0 +1,376 @@
+
+    @HttpMocker()
+    def test_given_slice_range_when_read_then_perform_multiple_requests(self, http_mocker: HttpMocker) -> None:
+        start_date = _NOW - timedelta(days=30)
+        slice_range = timedelta(days=20)
+        slice_datetime = start_date + slice_range
+
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _application_fees_request().with_created_gte(start_date).with_created_lte(slice_datetime).with_limit(100).build(),
+            _application_fees_response().build(),
+        )
+        http_mocker.get(
+            _application_fees_request().with_created_gte(slice_datetime + _AVOIDING_INCLUSIVE_BOUNDARIES).with_created_lte(_NOW).with_limit(100).build(),
+            _application_fees_response().build(),
+        )
+
+        self._read(_config().with_start_date(start_date).with_slice_range_in_days(slice_range.days))
+
+        # request matched http_mocker
+
+    @HttpMocker()
+    def test_given_http_status_400_when_read_then_stream_is_ignored(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            _application_fees_request().with_any_query_params().build(),
+            a_response_with_status(400),
+        )
+        output = self._read(_config())
+        assert len(output.get_stream_statuses(_STREAM_NAME)) == 0
+
+    @HttpMocker()
+    def test_given_http_status_401_when_read_then_system_error(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            _application_fees_request().with_any_query_params().build(),
+            a_response_with_status(401),
+        )
+        output = self._read(_config(), expecting_exception=True)
+        assert output.errors[-1].trace.error.failure_type == FailureType.system_error
+
+    @HttpMocker()
+    def test_given_rate_limited_when_read_then_retry_and_return_records(self, http_mocker: HttpMocker) -> None:
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _application_fees_request().with_any_query_params().build(),
+            [
+                a_response_with_status(429),
+                _application_fees_response().with_record(_an_application_fee()).build(),
+            ],
+        )
+        output = self._read(_config().with_start_date(_A_START_DATE))
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_http_status_500_once_before_200_when_read_then_retry_and_return_records(self, http_mocker: HttpMocker) -> None:
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _application_fees_request().with_any_query_params().build(),
+            [a_response_with_status(500), _application_fees_response().with_record(_an_application_fee()).build()],
+        )
+        output = self._read(_config())
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_http_status_500_on_availability_when_read_then_raise_system_error(self, http_mocker: HttpMocker) -> None:
```

Comment:
is this considered a bug? It's weird to me that we would retry on a 500 in the read but not in the check availability

Reply:
This test assumes only 500 HTTP statuses are returned by the server and does not validate if there are retries. I can confirm that there are retries. How can I change the naming to make sure things are clear?